lib/Parser/Parse.cpp

```diff
@@ -2252,7 +2301,26 @@ void Parser::AddModuleImportEntry(ModuleImportEntryList* importEntryList, IdentP
     importEntryList->Prepend(*importEntry);
 }
 
-void Parser::AddModuleExportEntry(ModuleExportEntryList* exportEntryList, IdentPtr importName, IdentPtr localName, IdentPtr exportName, IdentPtr moduleRequest)
+void Parser::CheckForDuplicateExportEntry(ModuleExportEntryList* exportEntryList, IdentPtr exportName)
+{
+    Js::PropertyId newPropertyId = EnsurePropertyId(exportName);
```

Comment:
Sounds good about the IdentPtr equality. Yes, export statements are only valid at global scope.